### PR TITLE
CVE-2018-1304

### PR DESCRIPTION
**Pull request by dungtran3864**
CVE-2018-1304

### CODE REVIEW

File: cves/CVE-2009-3548.yml [hunk 1]
```diff
@@ -43,29 +43,36 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  The Windows installer for Apache Tomcat uses a blank default 
+  password for the administrative user, which allows remote attackers to gain privileges easily.
+  This is obviously aggergious because it gives anyone admin access without having to
```
Comment (jbz6587) [inline cves/CVE-2009-3548.yml]: The correct spelling is egregious.

File: cves/CVE-2009-3548.yml [hunk 2]
```diff
@@ -139,12 +151,13 @@ interesting_commits:
     emerging themes?
     If there are no interesting commits, demonstrate that you completed this
     section by explaining what happened between the VCCs and the fix.
-  answer:
+  answer: |
+    Most of the commits are related to bug fixes with links to the bugs (9 in total). Others pertain to changing
+    defaults. The interesting thing here is that the original code written by rmaucher seems to have caused a lot
+    of issues. Given, it does seem to be a pretty important config file so that is to be expected.
```
Comment (jbz6587) [inline cves/CVE-2009-3548.yml]: What made it seem like rmaucher's changes caused a lot of these issues? Interesting maybe they were an intern ha-ha.

File: cves/CVE-2009-3548.yml [hunk 3]
```diff
@@ -183,8 +196,12 @@ lessons:
     applies:
     note:
   secure_by_default:
-    applies:
-    note:
+    applies: true
+    note: |
+      By default, the admin password should at least be specified for some amount of security.
+      If a user just clicks through midlessly during installation, the blank password will persist.
```
Comment (jbz6587) [inline cves/CVE-2009-3548.yml]: "... clicks through _mindlessly_ during..."

File: cves/CVE-2009-3548.yml [hunk 4]
```diff
@@ -202,4 +219,11 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    This appears to have been a design issue. The way the engineers designed the system, this vulnerability
+    was inherent. They didn't keep in mind the core security principle of secure by default. It wasn't until 
+    someone who was investigating security found this issue 3 years later that it was patched. It even mentioned 
+    in the response to the vulnerability report that the system behaved contrary to how it was described in the docs.
```
Comment (jbz6587) [inline cves/CVE-2009-3548.yml]: Surprised this issue was present for 3 years. What do you think caused this issue to not be noticed for this length of time? I recall you mentioned this issue had no prior or post fix.

File: cves/CVE-2009-3548.yml [hunk 5]
```diff
@@ -85,7 +92,7 @@ upvotes_instructions: |
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes:
+upvotes: 4
```
Comment (jbz6587) [inline cves/CVE-2009-3548.yml]: 3 upvotes.

File: cves/CVE-2011-1184.yml [hunk 6]
```diff
@@ -43,41 +43,47 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+    Attackers can look at all the messages being sent between the server and client, and get 
+    autheticated information. Then the attacker can use it by repeatedly sending the server the 
+    information. The server at this moment would authenticate the attacker because the message sent
+    has the correct username and password. The server should have seen the message was used already or 
+    old and dropped the message. In this case the server will give the attacker autheticated privlidges.
```
Comment (jbz6587) [inline cves/CVE-2011-1184.yml]: "...authenticated _privileges_."

File: cves/CVE-2011-1184.yml [hunk 7]
```diff
@@ -43,41 +43,47 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
```
Comment (jbz6587) [inline cves/CVE-2011-1184.yml]: Was there sensitive information being sent between the client and server? Or were these messages only used to obtain authentication?

File: cves/CVE-2011-1184.yml [hunk 8]
```diff
@@ -116,10 +127,16 @@ discovered:
     If there is no evidence as to how this vulnerability was found, then you
     may leave the entries blank except for "answer", BUT please write down
     where you looked in "answer".
-  answer:
-  date:
-  automated:
-  contest:
+  answer: |
+   The developers had found a separate weakness in the Tomcat's implementation for authentication 
+   about 3 months ago. They came out with a solution for that problem and when they
```
Comment (jbz6587) [inline cves/CVE-2011-1184.yml]: 3 months ago since when?

File: cves/CVE-2011-1184.yml [hunk 9]
```diff
@@ -116,10 +127,16 @@ discovered:
...
-  date:
-  automated:
-  contest:
+  answer: |
+   The developers had found a separate weakness in the Tomcat's implementation for authentication 
+   about 3 months ago. They came out with a solution for that problem and when they 
+   went to code review the fix, they found that there could be stale nonce values, and
+   the system would still accept them. Nonce is a value that will update as new messages are 
+   sent, so this can be used to see if a message is old or not. Which allows attackers to 
+   get elevated privilidges.
```
Comment (jbz6587) [inline cves/CVE-2011-1184.yml]: "...elevated _privileges_."

File: cves/CVE-2009-3548.yml (hunk 1 above)
Comment (CalvinWu4) [reply cves/CVE-2009-3548.yml]: Amazing how this got approved

File: cves/CVE-2009-3548.yml [hunk 10]
```diff
@@ -202,4 +219,11 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    This appears to have been a design issue. The way the engineers designed the system, this vulnerability
+    was inherent. They didn't keep in mind the core security principle of secure by default. It wasn't until 
+    someone who was investigating security found this issue 3 years later that it was patched. It even mentioned 
+    in the response to the vulnerability report that the system behaved contrary to how it was described in the docs.
+    The contrast between actual system behavior and the docs was clear midcommunication within the team.
```
Comment (CalvinWu4) [inline cves/CVE-2009-3548.yml]: Were there other security measures in place that prevented this vulnerability from being exploited?

File: cves/CVE-2012-0022.yml [hunk 11]
```diff
@@ -206,4 +220,11 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: From what I have seen, it seems like a design mistake led to this
+    vulnerability. It appears that when created, the original code did not
+    account for the possibly of large/excessive parameters.This could be due to
```
Comment (CalvinWu4) [inline cves/CVE-2012-0022.yml]: What do you mean by large/excessive parameters?

**Pull request merged**